backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -41,8 +41,9 @@ def collection
 
         @collection = super
         @search = @collection.ransack(params[:q])
-        @collection = @search.result(distinct: true).
+        @collection = @search.result.
```

Comment:
Crap, again my store not using multiple codes is biting me here as I'm not considering all the use cases. So basically:

* To allow searching by code we need the `distinct: true`.
* Having the `distinct: true` prevents us from sorting on custom criteria as anything in the `ORDER BY` must also be in the `SELECT`.

Option 1 - Something like `distinct: params[:q][:codes_value_cont].present?`. That way we are only doing `distinct` if searching on codes. Of course I would still need to resolve cases where the user is sorting on custom criteria but also searching on codes....
Option 2 - Drop this and the `promotion_references` completely as we get in a can of worms where one page has 10 extension points for every possible thing a developer might want to tweak. Are all those extension points more complex than just monkey-patching `collection`. Downside to this is I know solidus has a goal of discouraging monkey-patching.
Option 3 - Perhaps your `promotions_scope` hook could is the solution. Then I could also add a custom `select` that includes my custom sort (satisfying the SQL requirement that anything in `ORDER BY` also be in `SELECT` when using `DISTINCT`)

I think option 3 is the best. I can test that out to see if it works. Regarding monkey-patch vs config. If it was done as a config I wonder if we could do it at the Spree::ResourceController level. This way every resource controller has a hook to tweak it's scope to be customized. It could check for a config variable named after the controller name. If it doesn't exist then not tweaking needed.